core/services/workflows/syncer/workflow_registry.go

```diff
@@ -362,6 +355,192 @@ func (w *workflowRegistry) readRegistryEvents(ctx context.Context, don capabilit
+		})
+		if err != nil {
+			w.lggr.Errorf("failed to handle workflow registration: %s", err)
+		}
+	}
+
+	// Poll for all workflow related events
+	ets := []WorkflowRegistryEventType{
+		ForceUpdateSecretsEvent,
+		WorkflowActivatedEvent,
+		WorkflowDeletedEvent,
+		WorkflowPausedEvent,
+		WorkflowRegisteredEvent,
+		WorkflowUpdatedEvent,
+	}
+
+	w.readRegistryEvents(ctx, ets, don, reader, loadWorkflowsHead.Height)
+}
+
+// handleWorkflowMetadataAsEvents compares workflow metadata state against the engine registry's state.
+// Differences are handled by the event handler by creating events.
+func (w *workflowRegistry) handleWorkflowMetadataAsEvents(ctx context.Context, workflowMetadata []GetWorkflowMetadata, donID uint32) {
+	// Keep track of which of the engines in the engineRegistry have been touched
+	engineKeyMap := map[string]bool{}
+
+	for _, wfMeta := range workflowMetadata {
+		engine, getErr := w.engineRegistry.Get(EngineRegistryKey{Owner: wfMeta.Owner, Name: wfMeta.WorkflowName})
+
+		// if there is no engine in the engine registry
+		// then handle as registered event
+		if getErr != nil {
+			toRegisteredEvent := WorkflowRegistryWorkflowRegisteredV1{
+				WorkflowID:    wfMeta.WorkflowID,
+				WorkflowOwner: wfMeta.Owner,
+				DonID:         wfMeta.DonID,
+				Status:        wfMeta.Status,
+				WorkflowName:  wfMeta.WorkflowName,
+				BinaryURL:     wfMeta.BinaryURL,
+				ConfigURL:     wfMeta.ConfigURL,
+				SecretsURL:    wfMeta.SecretsURL,
+			}
+			err := w.handler.Handle(ctx, workflowAsEvent{
+				Data:      toRegisteredEvent,
+				EventType: WorkflowRegisteredEvent,
+			})
+			if err != nil {
+				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowRegisteredEvent)
+				continue
+			}
+
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, but the metadata has changed
+		// then handle as updated event
+		prevWfID := hex.EncodeToString(engine.workflowID[:])
+		currWfID := hex.EncodeToString(wfMeta.WorkflowID[:])
+		if currWfID != prevWfID {
+			toRegisteredEvent := WorkflowRegistryWorkflowUpdatedV1{
+				OldWorkflowID: engine.workflowID,
+				NewWorkflowID: wfMeta.WorkflowID,
+				WorkflowOwner: wfMeta.Owner,
+				DonID:         wfMeta.DonID,
+				WorkflowName:  wfMeta.WorkflowName,
+				BinaryURL:     wfMeta.BinaryURL,
+				ConfigURL:     wfMeta.ConfigURL,
+				SecretsURL:    wfMeta.SecretsURL,
+			}
+			err := w.handler.Handle(ctx, workflowAsEvent{
+				Data:      toRegisteredEvent,
+				EventType: WorkflowUpdatedEvent,
+			})
+			if err != nil {
+				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowUpdatedEvent)
+				continue
+			}
+
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, the metadata is the same, and the status is active
+		// then the workflow is unchanged. mark as seen.
+		if wfMeta.Status == 1 {
```

Comment:
Do we have a constant we can rely on for this status rather than using a magic number?

Reply:
Unfortunately not from the contract wrapper. There is one in /deployment in support of the view, but we don't have a clean pattern to re-use types between these.

I set up a new const for this file for now.